src/ai/backend/client/cli/params.py

```diff
@@ -166,3 +168,18 @@ def convert(self, arg, param, ctx):
                 return arg.split(",")
         except ValueError as e:
             self.fail(repr(e), param, ctx)
+
+
+class OptionalType(click.ParamType):
+    name = "Optional Type Wrapper"
+
+    def __init__(self, type_: type) -> None:
+        super().__init__()
+        self.type_ = type_
+
+    def convert(self, value: Any, param, ctx):
+        try:
+            if isinstance(value, self.type_) or value is undefined:
+                return value
+        except ValueError:
+            self.fail(f"{value!r} is not valid `{self.type_}` or `undefined`", param, ctx)
```

Comment:
```suggestion
    def convert(self, value: Any, param, ctx):
        try:
            if value is None or value is undefined:
                return value
            return self.type_(value)
        except ValueError:
            self.fail(f"{value!r} is not valid `{self.type_}` or `undefined`", param, ctx)
```
We should handle the case if `value` is not `type_` type instead of returning `None`.